nix/status-go/mobile/default.nix

```diff
@@ -1,19 +1,30 @@
 { callPackage, meta, source, goBuildLdFlags }:
 
-{
-  android = callPackage ./build.nix {
+let
+  android = {abis ? [ "armeabi-v7a" "arm64-v8a" "x86" ]}: callPackage ./build.nix {
     platform = "android";
     platformVersion = "23";
-    targets = [ "android/arm" "android/arm64" "android/386" ];
     outputFileName = "status-go-${source.shortRev}.aar";
+    # Hide different arch naming in gomobile from Android builds.
+    targets = let
+      abiMap = {
+        "armeabi-v7a" = "android/arm";
+        "arm64-v8a" = "android/arm64";
+        "x86" = "android/386";
+        "x86_64" = "android/386";
+        };
+      in map (arch: abiMap."${arch}") abis;
     inherit meta source goBuildLdFlags;
   };
 
   ios = callPackage ./build.nix {
```

Comment:
If we're doing this nice mapping for Android, we might as well do it for iOS. We don't have to use it in CI builds for it to be useful.